app/org/sagebionetworks/bridge/dynamodb/DynamoUploadDedupeDao.java

```diff
@@ -0,0 +1,63 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import javax.annotation.Resource;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import org.joda.time.LocalDate;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.BridgeConstants;
+import org.sagebionetworks.bridge.dao.UploadDedupeDao;
+import org.sagebionetworks.bridge.schema.UploadSchemaKey;
+
+/** DDB implementation of UploadDedupeDao. */
+@Component
+public class DynamoUploadDedupeDao implements UploadDedupeDao {
+    private static final int CREATED_ON_DELTA_MILLIS = 1000;
+
+    private DynamoDBMapper mapper;
+
+    /** UploadDedupe DDB mapper. */
+    @Resource(name = "uploadDedupeDdbMapper")
+    public void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isDuplicate(long createdOn, String healthCode, UploadSchemaKey schemaKey) {
+        // Hash key comes from health code and schema
+        DynamoUploadDedupe hashKey = new DynamoUploadDedupe();
+        hashKey.setHealthCode(healthCode);
+        hashKey.setSchemaKey(schemaKey.toString());
+
+        // range key, anything within 1 second is valid
+        Condition createdOnCondition = new Condition().withComparisonOperator(ComparisonOperator.BETWEEN)
+                .withAttributeValueList(
+                        new AttributeValue().withN(String.valueOf(createdOn - CREATED_ON_DELTA_MILLIS)),
+                        new AttributeValue().withN(String.valueOf(createdOn + CREATED_ON_DELTA_MILLIS)));
```

Comment:
Feedback from researchers is that the apps will sometimes upload the exact same set of files with a different createdOn timestamp for whatever reason. MD5 doesn't work, since if the createdOn is different, info.json will be different, and hence the MD5 will be different.

Based on discussions with the researcher, it's generally impossible for a user to complete the same survey or same activity more than once in a second, so keying off schema and createdOn should allow us to dedupe. The only concern is passive data collection, but if passive data is sending multiple results in a short time frame, it's almost certainly a bug in the client.